shell/common/engine.cc

```diff
@@ -403,35 +401,15 @@ bool Engine::HandleLocalizationPlatformMessage(
   return true;
 }
 
-bool Engine::HandleSystemPlatformMessage(blink::PlatformMessage* message) {
+void Engine::HandleSettingsPlatformMessage(blink::PlatformMessage* message) {
```

Comment:
Please avoid the JSON boxing/unboxing just to traverse a language boundary. You can use the standard binary encoding. You can use the same channel and disambiguate the setters by using the "type" field already present.

Reply:
Our binary codec lives in flutter/flutter. Moving it into the engine would be way out of scope for this PR. Notice that JSON is already used for this data, I'm not adding new JSON support. However, I'm willing to switch to binary, just not as part of this PR.

I added a second channel to remove a hack previously present in the code, which would fan out system messages based on type and values. This way all "settings" messages are handled in one place, and all "system" messages are handled in their own way without extra fanning out logic. That's what channels are for. Instead of multiplexing different kinds of data over one channel, use multiple channels.